Fix cursor positioning for the Virtual Keyboard

Districh-ru opened the pull request:
This PR fixes setting the cursor position by mouse (or screen tap) for the Virtual Keyboard and also fixes lag in screen update when doing this positioning.

oleg-derevenetz commented:
Hi @Districh-ru regarding the `renderInputArea()`: there is also another suspicious place - `KeyboardRenderer::resize()` line 170, should there be also used `_output.render( renderInputArea() );` instead of just `renderInputArea();`? If so, then I suggest to start with changing the name for this method, because for an inexperienced person, it may really seem like it performs rendering to the output. Also I can suggest to add the `[[nodiscard]]` attribute to it so the compiler warns when its return value is discarded.

Districh-ru commented:
Hi @oleg-derevenetz, the `KeyboardRenderer::resize()` does not need to render image on the display because it is done later in the caller function. And the returned `fheroes2::Rect` is not needed in this method so I did not add the `[[nodiscard]]` attribute.
I've renamed the `renderInputArea()` method to `_redrawInputArea()` and added a new method to redraw the input area.

oleg-derevenetz commented:
@Districh-ru Thank you very much!
